src/statement.ts

```diff
@@ -216,19 +221,21 @@ export class Statement<S = unknown> {
             type,
             tables,
           });
-          const results = await queryFirst<T>(config, sql, opts);
+          const results = await queryFirst<T>(config, sql, controller);
           if (results == null || colName == null) {
             return results;
           }
           return extractColumn(results, colName);
         }
         default: {
-          await this.#waitExec({
-            ...opts,
-            type,
-            sql,
-            tables,
-          });
+          await this.#waitExec(
+            {
+              type,
+              sql,
+              tables,
+            },
+            controller
+          );
```

Comment:
The old `opts` var was being injected into the statement object, never to be seen again. Pulling the controller out into its own var makes it easier to see what's going on.